### PR TITLE
refactor edgemesh and fix tcp test problem

### PR DESCRIPTION
**Pull request by liuzhiyi1993**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug

**What this PR does / why we need it**:
Fix some problems of edgemesh module:
1. In registry.go, the "port" of "EndpointMap" would now match the TCP port of the service.
2. Tests in tcp_test.go would now pass.
3. Also small refactor for resolver part.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: edgemesh/pkg/common/http_utils.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"strings"
+)
+
+func httpMethods() (methods []string) {
+	methods = []string{"GET", "HEAD", "POST", "OPTIONS", "PUT", "DELETE", "TRACE", "CONNECT"}
+	return
+}
+
+func IsHTTPRequest(s string) bool {
+	methods := httpMethods()
```
Comment (ugvddm) [inline edgemesh/pkg/common/http_utils.go]: why we need function call here? It's better to define a constant variable.

File: edgemesh/pkg/common/utils.go [hunk 2]
```diff
@@ -22,3 +24,13 @@ func SplitServiceKey(key string) (name, namespace string) {
 	}
 	return key, ns
 }
+
+func SplitToGetPort(serviceName string) (port int, err error) {
```
Comment (anyushun) [inline edgemesh/pkg/common/utils.go]: It's best to combine the two functions: ```SplitToGetPort``` and ```SplitServiceKey``` together:
```func ParseServiceName(serviceName string) (service, namespace, serviceType, cluster string, port int)```

File: edgemesh/pkg/resolver/http_resolver.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+				} else {
+					return invocation.Invocation{}, false
+				}
+			}
+			content += strData
+			req, err := http.ReadRequest(bufio.NewReader(bytes.NewReader([]byte(content))))
+			if err == nil {
+				content = ""
+				req.RequestURI = ""
+				i := invocation.New(context.Background())
```
Comment (anyushun) [inline edgemesh/pkg/resolver/http_resolver.go]: This function is too large, create a function called NewInvocation that returns an invocation please

File: edgemesh/pkg/registry/registry.go [hunk 4]
```diff
@@ -54,20 +41,35 @@ func (r *ServiceDiscovery) GetAllMicroServices() ([]*registry.MicroService, erro
...
+
 	var microServiceInstance []*registry.MicroServiceInstance
-	for _, p := range pods {
-		microServiceInstance = append(microServiceInstance, &registry.MicroServiceInstance{
-			InstanceID:   "",
-			ServiceID:    name + "." + namespace,
-			HostName:     "",
-			EndpointsMap: map[string]string{"rest": fmt.Sprintf("%s:%d", p.Status.HostIP, p.Spec.Containers[0].Ports[0].HostPort)},
-		})
+	for _, port := range service.Spec.Ports {
```
Comment (anyushun) [inline edgemesh/pkg/registry/registry.go]: Need update edgemesh end to end test guide  as well

**Comment by sids-b** [comment]
@liuzhiyi1993 , any progress on the review comments ?

**Comment by liuzhiyi1993** [comment]
@sids-b Hey Sid, I plan to fix these this weekend.

**Comment by sids-b** [comment]
please let me know after the comments are resolved. Thanks

**Pull request closed without merging**